### PR TITLE
switch arg name for metadata value class

### PR DESCRIPTION
**Pull request by smackesey**
This adjusts the new `MetadataEntry` class to (temporarily) accept either `value` or `entry_data` for its payload arg. In 0.15.0 will only be `value`.

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #6675** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/6675" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

**Review by sryza — CHANGES_REQUESTED** [review]
as discussed over slack, this could break storage and is worth reconsidering

**Comment by smackesey** [comment]
@sryza Having thought about this some more, handled properly this won't break storage. We can just use custom serialization logic to map older `entry_data` fields to `value` in 0.15.0. Given that, I think this diff is important to make a clean break with the "entry data" terminology.

**Review by sryza — COMMENTED** [review]
We're ultimately going to deprecate this class, right? If that's the case, it might make things easier just to leave things be?

**Comment by smackesey** [comment]
@sryza I didn't think the class was being deprecated-- just deprecating `description`. We're recommending users avoid instantiating `MetadataEntry` directly, but internally everything is passed around as lists of `MetadataEntry`, and per previous discussion I thought that was not going to change.

**Review by sryza — COMMENTED** [review]
This makes sense to me, but I think I'm relatively blind to the difficulties (or lack there-of) with storage back-compatibility, so I'd like @alangenfeld to take a look as well.

E.g. are there issues here with storage _forward_-compatibility?

**Comment by smackesey** [comment]
> E.g. are there issues here with storage forward-compatibility?

Could you explain a little what you mean by this? My interpretation of "forward-compatibility" is: will this format limit our ability to change how these values are persisted in the future? If that's what you're asking, then I don' think so.

File: python_modules/dagster/dagster/core/definitions/metadata/__init__.py [hunk 1]
```diff
@@ -80,45 +86,50 @@ def normalize_metadata(
...
+        raise DagsterInvalidMetadata(
+            f"Expected a metadata value, found an instance of {raw_value.__class__.__name__}. Consider "
+            "instead using a MetadataValue wrapper for the value."
+        )
+    try:
+        value = normalize_metadata_value(raw_value)
+    except DagsterInvalidMetadata as e:
+        raise DagsterInvalidMetadata(
+            f'Could not resolve the metadata value for "{label}" to a known type. {e}'
+        ) from None
```
[the remark this answers is not in the document]
Comment (smackesey) [reply python_modules/dagster/dagster/core/definitions/metadata/__init__.py:129]: This is used to [suppress error context](https://stackoverflow.com/a/24752607/1244455), it's useful if you want to clean up your error messages when you are catching one error and replacing it with another (especially of the same type, as I did here).

**Review by alangenfeld — APPROVED** [review]
seems fine to me

adding some tests not a bad idea

**Pull request merged**